test/healthcheck/HealthChecker.test.ts

```diff
@@ -1145,4 +1145,36 @@ describe('Health Checker test suite', () => {
     const expected = "{\"status\":\"STOPPING\",\"checks\":[{\"name\":\"checkone\",\"state\":\"STOPPED\",\"data\":{\"reason\":\"\"}},{\"name\":\"checktwo\",\"state\":\"STOPPING\",\"data\":{\"reason\":\"\"}}]}"
     expect(result).to.equal(expected, `Should return: ${expected}, but returned: ${result}`);
   });
+
+  it('Readiness should return DOWN with a failing Readiness check not throwing an error' , async() => {
+    let healthcheck = new HealthChecker();
+    const readinessPromise = () => new Promise<void>((resolve, reject) => {
+      reject("Readiness Failure");
+    });
+
+    let check = new ReadinessCheck("readinessCheck", readinessPromise);
+    healthcheck.registerReadinessCheck(check);
+    let status = await healthcheck.getReadinessStatus();
+    status = await healthcheck.getReadinessStatus();
+    const result = JSON.stringify(status);
+
+    let expected = "{\"status\":\"\DOWN\",\"checks\":[{\"name\":\"readinessCheck\",\"state\":\"DOWN\",\"data\":{\"reason\":\"Readiness Failure\"}}]}";
+    expect(result).to.equal(expected, `Should return: ${expected}, but returned: ${result}`);
+  });
+
+  it('Readiness should return DOWN with a failing Readiness check not throwing an error with no message' , async() => {
```

Comment:
I might be lost in the double negative, but I think the below *is* rejecting with an error (`undefined`) with no message? This is testing "reject with no explicit argument", it looks to me.